### PR TITLE
Update docs to latest version

### PR DESCRIPTION
**Pull request by davidmirror-ops**
Signed-off-by: David Espejo <daespejo@vmware.com>

Also in this PR, I created a `main` folder for unreleased version docs

Your comments/thoughts/reviews are welcome

### CODE REVIEW

File: site/config.yaml [hunk 1]
```diff
@@ -57,8 +57,10 @@ params:
   docs_search_index_name: index_name
   docs_search_api_key: api_key
   docs_versioning: true
-  docs_latest: v0.0.6
+  docs_latest: v0.0.7
   docs_versions:
+    - main
```
Comment (cirocosta) [inline site/config.yaml]: wdyt of `latest`? I think `main` indeed makes sense from the perspective of "versioning is based on git references", but when you look at the website, I think the concept of "latest version" .. "version 0.0.6 .." makes more sense - wdyt? 

---

(regardless, great to now be able to select the most up to date version!)

File: site/data/docs/main-toc.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  - title: Examples
+    subfolderitems:
+      - page: Examples
+        url: /examples
+  - title: Documentation Style Guide
+    subfolderitems:
+      - page: Style Guide
+        url: /style-guide
+
+  #     - page: Scope 2
```
Comment (cirocosta) [inline site/data/docs/main-toc.yml]: looking at the others, it seems like this came straight from what the others had too - should we clean it from this new one right now or do you think we should do some cleanup in a separate PR?

File: site/config.yaml (hunk 1 above)
Comment (cirocosta) [reply site/config.yaml]: feedback from community call: 

- `latest` could instead be used as a pointer to the latest released version (analogous to how https://github.com/vmware-tanzu/cartographer/releases/latest points at whatever latest released tag we have)
- maybe call this .. `preview`? maybe `develpment`?

Comment (davidmirror-ops) [reply site/config.yaml]: My vote is for `development` as it is clear for users that it refers to unreleased software on which the team is actively working. For me as user, `latest` refers to the most recently released and stable version, and `preview` sounds like it will eventually see the light but not sure when :)

Please upvote your options, as soon as we have consensus, I can update the PR

File: site/data/docs/main-toc.yml (hunk 2 above)
Comment (davidmirror-ops) [reply site/data/docs/main-toc.yml]: yes, it's a copy from the other tocs. Why should we clean this one? Do we want to have a different ToC for the unreleased version?

File: site/config.yaml (hunk 1 above)
Comment (emmjohnson) [reply site/config.yaml]: @squeedee what did we come up with in the docs meeting?

Comment (squeedee) [reply site/config.yaml]: "Edge" IIRC.

Comment (squeedee) [reply site/config.yaml]: 100% agree "Latest" is latest-known-good.. I'm also ok with "Development"

Comment (emmjohnson) [reply site/config.yaml]: Another favorite from community meeting was Development (putting here so people can vote)

Comment (cirocosta) [reply site/config.yaml]: +1 on "development", and "latest" for pointing at the latest release (if that's possible)

File: site/data/docs/main-toc.yml (hunk 2 above)
Comment (cirocosta) [reply site/data/docs/main-toc.yml]: oh noo, no worries, just trying to figure out whether the comments were something we wanted to clean up in this PR or elsewhere (given that, being comments, they seem like leftovers from experimenting at some point)

Comment (davidmirror-ops) [reply site/data/docs/main-toc.yml]: I'm kinda used to addressing comments in the same PR but not an expert here so, the idea would be to have a separate PR addressing how to handle unreleased vs latest-stable naming only?

Comment (cirocosta) [reply site/data/docs/main-toc.yml]: I personally would go for a separate one where we get rid of the commented sections, but no hard opinions on my side 😁  if you feel like deleting here in this one, all good!

File: site/config.yaml (hunk 1 above)
Comment (davidmirror-ops) [reply site/config.yaml]: Updated PR with "development" for unreleased version docs. Using `latest` along the most recent version number would add at least two additional steps to the process whenever we need to update for a new release

**Comment by cirocosta** [comment]
lgtm! there are a couple conflicts at the moment, but that's really it

```
site/content/docs/_index.md
site/content/docs/about.md
site/content/docs/install.md
site/content/docs/reference.md
```

are you familiar with how to resolve those, @davidmirror-ops ? I think those conflicts are coming mostly from #344. if you need any help, just let me know!

thanks!

**Comment by davidmirror-ops** [comment]
Hey @cirocosta! Well not sure how to fix those, but I can try (so I learn) and if I can't, will request help :)

**Comment by davidmirror-ops** [comment]
soo I tried it with a draft PR but ended up with repeated commits in the log so it seems I need help :)

**Comment by squeedee** [comment]
Closing with [#403 as replacement](https://github.com/vmware-tanzu/cartographer/pull/403)
